freq/views.py

```diff
@@ -1,283 +1,913 @@
-        self.set_session(request)
-        return super().get(request, *args, **kwargs)
+
+class StartPageView(StartPageBaseView):
+    active = 'startpage'
+    freq_active = 'active'
+    template_name = 'freq/startpage.html'
+
+    def dispatch(self, *args, **kwargs):  # TODO: PUT IN ALL VIEWS
+        if not self.request.session.get('session_is_set', False):
+            self.instantiate_session()
+        return super().dispatch(*args, **kwargs)
 
 
 class ReStartPageView(StartPageView):
 
     def get(self, request, *args, **kwargs):
-        self.flush(request)
-        self.set_session(request)
+        self.instantiate_session()
         return super().get(request, *args, **kwargs)
 
 
-class TimeSeriesQueryView(StartPageView):
-
-    def get(self, request, x_coord, y_coord, *args, **kwargs):
-        kwargs['multiple_timeseries'] = False
-        ts = TimeSeries()
-        ts.location_uuid(kwargs['uuid'])
-        request.session['startpage']['selected_coords'] = [x_coord, y_coord]
-        if len(ts.results) == 0:
-            kwargs['error_message'] = 'No time series found for this ' \
-                                      'location, please select another.'
-        elif len(ts.results) == 1:
-            request.session['startpage']['measurement_point'] = \
-                "Meetpunt: " + ts.results[0]['name']
-            request.session['disabled']['trend_detection'] = ''
-            request.session['trend_detection'] = {'active':True}
-        else:
-            kwargs['error_message'] = 'Multiple time series found for ' \
-                                      'this location, please select one ' \
-                                      'below.'
-            kwargs['multiple_timeseries'] = True
-            kwargs['timeseries'] = [
+class TimeSeriesByLocationUUIDView(StartPageBaseView):
+    error_message = ''
+
+    @cached_property
+    def timeseries(self):
+        ts = GroundwaterTimeSeries()
+        ts.location_uuid(self.uuid)
+        return ts.results
+
+    @cached_property
+    def multiple_timeseries(self):
+        return len(self.timeseries) > 1
+
+    @cached_property
+    def error_message(self):
+        if self.multiple_timeseries:
+            return 'Multiple time series found for this location, please ' \
+                   'select one below.'
+        elif len(self.timeseries) == 1:
+            return 'No time series found for this location, please select ' \
+                   'another.'
+        return ''
+
+    @property
+    def timeseries_selection(self):
+        if self.multiple_timeseries:
+            return [
                 (
                     x['location']['name'] + ' - ' + x['name'],
                     x['uuid'],
                     x['first_value_timestamp'],
                     x['last_value_timestamp']
                 )
-                for x in ts.results]
+                for x in self.timeseries
+            ]
+        elif len(self.timeseries) == 1:
```

Comment:
Dit mag ook een reguliere "if" zijn.

En misschien `if not len(...): return []` hier doen en dan de rest van de code die hier staat vier spaties naar rechts? De "common case" staat idealiter in de hoofdstroom van het programma. De uitzonderingen in de "if". Zoiets.